src/Backend/Core/Engine/DataGrid.php

```diff
@@ -367,70 +368,70 @@ public function setColumnConfirm($column, $message, $custom = null, $title = nul
-                        $value = str_replace(
-                            'class="',
-                            'data-message-id="' . $id . '" class="jsConfirmationTrigger ',
-                            $value
-                        );
-                    } else {
-                        $value = str_replace(
-                            '<a ',
-                            '<a data-message-id="' . $id . '" class="jsConfirmationTrigger" ',
-                            $value
-                        );
-                    }
-                } else {
-                    // is it a link?
-                    throw new Exception('The column doesn\'t contain a link.');
-                }
-
-                // append message
-                $value .= '<div id="' . $id . '" title="' . $title . '" style="display: none;"><p>' .
-                          $message . '</p></div>';
-
-                // reset value
-                $this->columns[$column]->setValue($value);
+            // set title if there wasn't one provided
+            if ($title === null) {
+                $title = \SpoonFilter::ucfirst(BackendLanguage::lbl('Delete') . '?');
+            }
+
+            // grab current value
+            $value = $this->columns[$column]->getValue();
+
+            // add class for confirmation
+            if (mb_substr_count($value, '<a') === 0) {
+                // is it a link?
+                throw new Exception('The column doesn\'t contain a link.');
             }
+
+            if (mb_substr_count($value, 'class="') > 0) {
+                $value = str_replace(
+                    'class="',
+                    'data-message-id="' . $id . '" class="jsConfirmationTrigger ',
+                    $value
+                );
+            } else {
+                $value = str_replace(
+                    '<a ',
+                    '<a data-message-id="' . $id . '" class="jsConfirmationTrigger" ',
+                    $value
+                );
+            }
+
+            // append message
+            $value .= '<div id="' . $id . '" title="' . $title . '" style="display: none;"><p>' .
+                      $message . '</p></div>';
+
+            // reset value
+            $this->columns[$column]->setValue($value);
         }
     }
 
     /**
      * Sets the column function to be executed for every row
      *
-     * @param mixed $function  The function to execute.
+     * @param mixed $function The function to execute.
      * @param mixed|null $arguments The arguments to pass to the function.
-     * @param mixed $columns   The column wherein the result will be printed.
-     * @param bool  $overwrite Should the original value be overwritten.
+     * @param string[]|string $columns The column wherein the result will be printed.
+     * @param bool $overwrite Should the original value be overwritten.
      */
     public function setColumnFunction($function, $arguments, $columns, $overwrite = true)
```

Comment:
only in the php doc, did that in a60eb7fee
the reason we can't do that is because it needs to match the definition of the same function in SpoonLibrary

Maybe we can do that in the PR to clean up SpoonLibrary but not at the moment